sys/net/gnrc/link_layer/netdev2/gnrc_netdev2.c

```diff
@@ -0,0 +1,209 @@
+    dev->isr_arg = (void*) gnrc_netdev2;
+
+    /* register the device to the network stack*/
+    gnrc_netif_add(thread_getpid());
+
+    /* initialize low-level driver */
+    dev->driver->init(dev);
+
+    /* start the event loop */
+    while (1) {
+        DEBUG("gnrc_netdev2: waiting for incoming messages\n");
+        msg_receive(&msg);
+        /* dispatch NETDEV and NETAPI messages */
+        switch (msg.type) {
+            case NETDEV2_MSG_TYPE_EVENT:
+                DEBUG("gnrc_netdev2: GNRC_NETDEV_MSG_TYPE_EVENT received\n");
+                dev->driver->isr(dev);
+                break;
+            case GNRC_NETAPI_MSG_TYPE_SND:
+                DEBUG("gnrc_netdev2: GNRC_NETAPI_MSG_TYPE_SND received\n");
+                gnrc_pktsnip_t *pkt = (gnrc_pktsnip_t *)msg.content.ptr;
+                gnrc_netdev2->send(gnrc_netdev2, pkt);
+                break;
+            case GNRC_NETAPI_MSG_TYPE_SET:
+                /* read incoming options */
+                opt = (gnrc_netapi_opt_t *)msg.content.ptr;
+                DEBUG("gnrc_netdev2: GNRC_NETAPI_MSG_TYPE_SET received. opt=%s\n",
+                        netopt2str(opt->opt));
+                /* set option for device driver */
+                res = dev->driver->set(dev, opt->opt, opt->data, opt->data_len);
+                DEBUG("gnrc_netdev2: response of netdev->set: %i\n", res);
+                /* send reply to calling thread */
+                reply.type = GNRC_NETAPI_MSG_TYPE_ACK;
+                reply.content.value = (uint32_t)res;
+                msg_reply(&msg, &reply);
+                break;
+            case GNRC_NETAPI_MSG_TYPE_GET:
+                /* read incoming options */
+                opt = (gnrc_netapi_opt_t *)msg.content.ptr;
+                DEBUG("gnrc_netdev2: GNRC_NETAPI_MSG_TYPE_GET received. opt=%s\n",
+                        netopt2str(opt->opt));
+                /* get option from device driver */
+                res = dev->driver->get(dev, opt->opt, opt->data, opt->data_len);
+                DEBUG("gnrc_netdev2: response of netdev->get: %i\n", res);
+                /* send reply to calling thread */
+                reply.type = GNRC_NETAPI_MSG_TYPE_ACK;
+                reply.content.value = (uint32_t)res;
+                msg_reply(&msg, &reply);
+                break;
+            default:
+                DEBUG("gnrc_netdev2: Unknown command %" PRIu16 "\n", msg.type);
+                break;
+        }
+    }
+    /* never reached */
+    return NULL;
+}
+
+kernel_pid_t gnrc_netdev2_init(char *stack, int stacksize, char priority,
+                        const char *name, gnrc_netdev2_t *gnrc_netdev2)
+{
+    kernel_pid_t res;
+
+    /* check if given netdev device is defined and the driver is set */
+    if (gnrc_netdev2 == NULL || gnrc_netdev2->dev == NULL) {
+        return -ENODEV;
+    }
+
+    /* create new gnrc_netdev2 thread */
+    res = thread_create(stack, stacksize, priority, CREATE_STACKTEST,
```

Comment:
All I'm saying is: we need to make 100% clear - so that a new GNRC contributor sees it at first (or maybe at least second) glance - that this is thought to replace `gnrc_nomac`.